pallets/creditcoin/src/tests.rs

```diff
@@ -49,29 +51,71 @@ pub struct RegisteredAddress {
 	account_id: AccountId,
 }
 impl RegisteredAddress {
+	pub fn from_pubkey_distinct_owner(
+		owners_account_id: AccountId,
+		blockchain: Blockchain,
+		address_key: impl Into<MultiSigner>,
+		signature: sp_core::ecdsa::Signature,
+	) -> RegisteredAddress {
+		let signer = address_key.into();
+		let address = if let MultiSigner::Ecdsa(pkey) = signer.clone() {
+			let address = Etherlike::to_checksum_address(Etherlike::from_public(pkey));
+			format!("0x{}", hex::encode(address)).hex_to_address()
```

Comment:
This is effectively a no-op. Checksumming doesn't have any effect because the case-based checksum is a property of the hex/ascii encoding of an address, not a property of the address itself. As soon as you decode a hex-encoded address into its actual value (which is what `hex_to_address` does, it takes the hex-encoded address and decodes it into the actual bytes of the address), you lose that property. See this test:
```rust
#[test]
fn checksumming_and_re_encoding_addr_is_a_noop() {
	let (key_pair, _) = sp_core::ecdsa::Pair::generate();
	let pkey = key_pair.public();
	let address = Etherlike::from_public(pkey);
	let chk_address = Etherlike::to_checksum_address(address.clone());
	let chk_re_encoded = format!("0x{}", hex::encode(chk_address)).hex_to_address();
	assert_eq!(address, chk_re_encoded);
}
```

Reply:
Yup, It is a no-op. 
> Checksumming doesn't have any effect because the case-based checksum is a property of the hex/ascii encoding of an address, not a property of the address itself.

I don't agree, it is a property of the operators where $h(f(g(x))) = x$ and $g(x) != x$. During the transformation, the bytes are lowercased in the process for convenience, I would say the actual bytes are whatever got in, not what got out. 
I kept the no-op because I tried to keep as many pre and post conditions as possible, in some sense, it is more correct to me. I don't see anywhere stated that $h(f(g(x))) = x$ is an identity operation so I didn't assume that even though I knew. The hardcoded addresses had capitals, in the old test it didn't matter, due to the operator. The new impl still has uppercase and it still doesn't matter due to the operators. Up to QA to choose what to do, as an implementer I tried to preserve the tests correctness. There is a bunch of repetitive code and optimizations that can be done. Including the one you pointed out. I may be saying nonsense but this was my reasoning.

But I'm on your side, I suggest including your test and getting rid of all the no-ops scattered around. I wear my QA hat while testing, not that I'm good at it smh.